pagopa/io-app
# feat: [IOBP-514] Move authorization creation request upon payment method selection

## Short description
This PR moves the transaction creation request from the payment summary screen (`WalletPaymentConfirmScreen`) to the payment method selection (`WalletPaymentPickMethodScreen`), after the payment method selection.

## List of changes proposed in this pull request
- Refactored `walletPaymentCreateTransaction` action to accept an `onSuccess` callback
- Added `walletPaymentCreateTransaction` dispatch in `WalletPaymentPickMethodScreen`, after _Continue_ button press

## How to test
With the [io-dev-api-server,](https://github.com/pagopa/io-dev-api-server) start a new payment flow from the **Profile > Playgrounds > New Wallet > Payment** screen. With reactotron, check that the transaction creation request is dispatched before the PSP selection screen.
